src/lib/apparmor/logprof.rb

```diff
@@ -0,0 +1,87 @@
+
+require 'open3'
+require 'json'
+require 'yast'
+require 'apparmor/apparmor_ui_dialog'
+
+Yast.import 'UI'
+Yast.import 'Label'
+Yast.import 'Popup'
+
+module AppArmor
+  # LogProf class executes the aa-logprof command and generates
+  # the appropriate dialogs for the yast UI
+  class LogProf
+    LOGPROF = '/usr/sbin/aa-logprof'.freeze
+    attr_reader :logfile
+    include Yast::UIShortcuts
+    include Yast::Logger
+    include Yast::I18n
+
+    def initialize(logfile = '')
+      textdomain 'apparmor'
+      @logfile = logfile
+      log.info "Logfile is #{logfile}."
+    end
+
+    def execute
+      cmd = "#{LOGPROF} --json"
+      cmd += " -f #{@logfile}" unless @logfile.empty?
+      log.info "Executing #{cmd}"
+      IO.popen(cmd, 'r+') do |f|
+        f.sync = true
+        f.each do |line|
+          log.info "aa-logprof lines #{line}."
+          next unless line.start_with?('{')
+          hm = JSON.parse(line)
+          l = get_dialog(hm)
+          r = l.run
+          unless r.nil?
+            f.puts r.to_json
+            f.flush
+          end
+        end
```

Comment:
for me this part looks quite similar to GenProf, maybe have common ancestor that share it?